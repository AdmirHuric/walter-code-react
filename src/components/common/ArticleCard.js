import React from "react";
import {Button, Card} from "react-bootstrap";
import PropTypes from "prop-types";

const ArticlesCard = ({article, onArticleClick}) => (
  <Card className="shadow-sm">
    <Card.Header className="font-weight-bold">{article.source.name}</Card.Header>
    <Card.Body className="text-left">
      {
        article.urlToImage &&
        <Card.Img
          style={{maxHeight: 300}}
          className="card-img-bottom h-60" src={article.urlToImage}/>
      }
      <p className="font-weight-normal text-muted h-40 py-2">{article.description}</p>
    </Card.Body>
    <Card.Footer className="text-center">
      <Button onClick={onArticleClick.bind(this, article)}>Read Full Article</Button>
    </Card.Footer>
  </Card>
);

ArticlesCard.propTypes = {
  article: PropTypes.object.isRequired,
  onArticleClick: PropTypes.func.isRequired
}

export default ArticlesCard;
