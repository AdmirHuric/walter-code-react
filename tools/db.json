{"topHeadlines":{"status":"ok","totalResults":38,"articles":[{"source":{"id":"the-wall-street-journal","name":"The Wall Street Journal"},"author":"Charley Grant","title":"Cleared Hurdle Still Leaves Tesla Stock in Ludicrous Mode - The Wall Street Journal","description":"Tesla meets own guidance for annual production, but valuation still looks hard to justify","url":"https://www.wsj.com/articles/cleared-hurdle-still-leaves-tesla-stock-in-ludicrous-mode-11609622182","urlToImage":"https://images.wsj.net/im-280026/social","publishedAt":"2021-01-03T14:05:00Z","content":"Tesla announced on Saturday morning that it delivered 180,570 cars world-wide in the fourth quarter, setting a new company record. That brings the 2020 total just shy of 500,000, in line with the com… [+2719 chars]"},{"source":{"id":"cbs-news","name":"CBS News"},"author":"Jordan Freiman","title":"Nashville bomber sent writings espousing conspiracy theories to multiple people days before blast - CBS News","description":"Federal law enforcement confirmed the packages were postmarked December 23, just two days before the Christmas day explosion.","url":"https://www.cbsnews.com/news/nashville-bomber-anthony-quinn-sent-conspiracy-theories-to-people-before-explosion/","urlToImage":"https://cbsnews1.cbsistatic.com/hub/i/r/2020/12/26/8ddb84de-6988-46c5-a0c5-3ea11abf1465/thumbnail/1200x630/81dda50732b9d2974e244ca54c1ccecd/ap-20360586316342.jpg","publishedAt":"2021-01-03T12:41:00Z","content":"The Nashville bomber sent packages containing writings and videos promoting conspiracy theories to multiple people just days prior to the blast, CBS News confirms. Authorities identified Anthony Warn… [+1900 chars]"},{"source":{"id":null,"name":"Seeking Alpha"},"author":null,"title":"Bitcoin tops $34,000 as rally extends into 2021 - Seeking Alpha","description":"While most markets were closed over the first weekend of 2021, an eye-popping rally for Bitcoin (BTC-USD) continued.At the time of writing, Bitcoin is up 14.7% to $34,108.09, with more traders and institutional investors betting on the crypto becoming a mains…","url":"https://seekingalpha.com/news/3648195-bitcoin-tops-34000-rally-extends-2021","urlToImage":"https://static3.seekingalpha.com/assets/og_image_192-59bfd51c9fe6af025b2f9f96c807e46f8e2f06c5ae787b15bf1423e6c676d4db.png","publishedAt":"2021-01-03T12:40:00Z","content":null},{"source":{"id":"cbs-news","name":"CBS News"},"author":"CBS News","title":"Hospitals on high alert for new, more contagious COVID-19 strain - CBS News","description":"Florida has become the third state to report a case of the COVID-19 variant.","url":"https://www.cbsnews.com/news/new-covid-strain-contagious-hospitals-high-alert/","urlToImage":"https://cbsnews2.cbsistatic.com/hub/i/r/2021/01/02/7ecc7a4a-d4bf-48c5-8f51-7bbaa7226e96/thumbnail/1200x630/f4d678f72c4019cb0a0228040a5c723e/0102-satmo-covidwrap-hanson-618996-640x360.jpg","publishedAt":"2021-01-03T12:35:00Z","content":"Hospitals in the United States are on high alert for the new, more contagious COVID-19 strain that first surfaced in the United Kingdom and has since shown up in Colorado, California and now Florida.… [+2812 chars]"},{"source":{"id":null,"name":"Yahoo Entertainment"},"author":"Maher Nazeh","title":"Iraqis chanting anti-U.S. slogans mark year since Soleimani killing - Yahoo News","description":"BAGHDAD (Reuters) -Tens of thousands of supporters of Iranian-backed Iraqi paramilitary groups chanted anti-American slogans in central Baghdad on Sunday to ...","url":"https://news.yahoo.com/iraqis-chanting-anti-u-slogans-122308779.html","urlToImage":"https://s.yimg.com/uu/api/res/1.2/D6.eoLvUTvTtEBtEctj.ug--~B/aD01MzM7dz04MDA7YXBwaWQ9eXRhY2h5b24-/https://media.zenfs.com/en/reuters.com/f5cabcab902080d2f850b28539dcbfbd","publishedAt":"2021-01-03T12:23:00Z","content":"By Maher Nazeh\r\nBAGHDAD (Reuters) -Tens of thousands of supporters of Iranian-backed Iraqi paramilitary groups chanted anti-American slogans in central Baghdad on Sunday to mark the anniversary of th… [+2654 chars]"},{"source":{"id":"cnn","name":"CNN"},"author":"Alex Rogers and Lauren Fox, CNN","title":"Nancy Pelosi must count carefully ahead of House speakership vote Sunday - CNN","description":"House Speaker Nancy Pelosi has a slim margin for error on Sunday when lawmakers return to Washington to officially usher in a new session of Congress and elect their new speaker.","url":"https://www.cnn.com/2021/01/03/politics/nancy-pelosi-speaker-vote/index.html","urlToImage":"https://cdn.cnn.com/cnnnext/dam/assets/201119013958-nancy-pelosi-0822-super-tease.jpg","publishedAt":"2021-01-03T12:18:00Z","content":null}]}}