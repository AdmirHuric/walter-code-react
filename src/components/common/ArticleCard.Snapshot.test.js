import React from "react";
import ArticleCard from "./ArticleCard";
import renderer from "react-test-renderer";
import { topHeadlines } from "../../../tools/mockData";

it("renders correctly", () => {
  const tree = renderer.create(
    <ArticleCard
      article={topHeadlines.articles[0]}
      onArticleClick={jest.fn()}
    />
  );

  expect(tree).toMatchSnapshot();
});
