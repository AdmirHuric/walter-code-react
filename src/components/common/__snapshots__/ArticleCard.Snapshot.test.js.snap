// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="shadow-sm card"
>
  <div
    className="font-weight-bold card-header"
  >
    The Wall Street Journal
  </div>
  <div
    className="text-left card-body"
  >
    <img
      className="card-img card-img-bottom h-60"
      src="https://images.wsj.net/im-280026/social"
      style={
        Object {
          "maxHeight": 300,
        }
      }
    />
    <p
      className="font-weight-normal text-muted h-40 py-2"
    >
      Tesla meets own guidance for annual production, but valuation still looks hard to justify
    </p>
  </div>
  <div
    className="text-center card-footer"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Read Full Article
    </button>
  </div>
</div>
`;
